src/sentry/api/serializers/models/releaseactivity.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+from typing import Any, Mapping, MutableMapping, Sequence, TypedDict
+
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models import Group
+from sentry.models.releaseactivity import ReleaseActivity
+from sentry.types.releaseactivity import ReleaseActivityType
+from sentry.utils.json import JSONData
+
+
+class ReleaseActivityResponse(TypedDict):
+    id: int
+    type: str
+    data: Mapping[str, JSONData]
+    date_added: datetime
+
+
+@register(ReleaseActivity)
+class ReleaseActivitySerializer(Serializer):
+    def get_attrs(
+        self, item_list: Sequence[ReleaseActivity], user: Any, **kwargs: Any
+    ) -> MutableMapping[Any, Any]:
+        groups = list(
+            Group.objects.filter(
+                id__in=[
+                    i.data.get("group_id") for i in item_list if i.data and i.data.get("group_id")
+                ]
+            )
+        )
+        serialized_groups = {g["id"]: g for g in serialize(groups, user)}
+
+        def _expand_group(d: Mapping[str, Any]) -> Mapping[str, Any]:
+            if d.get("group_id"):
+                d_copy = dict(d)
+                d_copy["group"] = serialized_groups.get(str(d.get("group_id")), None)
+                return d_copy
+
+            return d
+
+        return {activity: _expand_group(activity.data) for activity in item_list}
+
+    def serialize(
+        self, obj: Any, attrs: Mapping[Any, Any], user: Any, **kwargs: Any
```

Comment:
Same here for the mapping

Reply:
Hmm, I think the `attrs` is the value of whatever is mapped from `get_attrs`? In this case, itll be the type defined by the `data` 'Mapping[str, JSONData]', right?

https://github.com/getsentry/sentry/blob/9113cb2b7db1b69a7e206d90df93a7404aa6c659/src/sentry/api/serializers/base.py#L117